dbt/include/databricks/macros/materializations/snapshot.sql

```diff
@@ -48,25 +16,6 @@
     {% do return(tmp_relation) %}
 {% endmacro %}
 
-
-{% macro databricks__post_snapshot(staging_relation) %}
-    {% do adapter.drop_relation(staging_relation) %}
-{% endmacro %}
-
-
-{% macro databricks__create_columns(relation, columns) %}
-    {% if columns|length > 0 %}
-    {% call statement() %}
-      alter table {{ relation }} add columns (
-        {% for column in columns %}
-          `{{ column.name }}` {{ column.data_type }} {{- ',' if not loop.last -}}
-        {% endfor %}
-      );
-    {% endcall %}
-    {% endif %}
-{% endmacro %}
-
-
 {% materialization snapshot, adapter='databricks' %}
```

Comment:
It looks like you discovered this yourselves, but just for the sake of stating clearly:

Materializations don't natively support the same sort of "parent" adapter inheritance as other macros. I'm going to open an issue in `dbt-core` that addresses this. In the meantime, you can "mock" that inheritance (and avoid duplicated code) via:
```sql
{% materialization snapshot, adapter='databricks' %}
    {# this actually calls the spark version of the snapshot materialization #}
    {% set relations = materialization_snapshot_spark() %}
    {{ return(relations) }}
{% endmaterialization %}
```
This isn't a capability we document or recommend for end users, but [we use it in `dbt-snowflake` as well](https://github.com/dbt-labs/dbt-snowflake/blob/main/dbt/include/snowflake/macros/materializations/snapshot.sql). Any better approach we implement in a new `dbt-core` minor version will aim to include a non-breaking upgrade path / backwards compatibility for this syntax.

Reply:
Thanks for letting me know. That's great to know.
Actually for this materialization, there are some updates for dbt-databricks from dbt-spark, so I think we still need to redefine here.